lib/astarte/core/generators/interface.ex

```diff
@@ -0,0 +1,169 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2025 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule Astarte.Core.Generators.Interface do
+  @moduledoc """
+  This module provides generators for Astarte Interface.
+
+  See https://docs.astarte-platform.org/astarte/latest/030-interface.html
+  """
+  use ExUnitProperties
+
+  alias Ecto.UUID
+  alias Astarte.Core.Interface
+  alias Astarte.Core.Generators.Mapping, as: MappingGenerator
+
+  @doc """
+  Generates a valid Astarte Interface.
+
+  https://github.com/astarte-platform/astarte_core/blob/master/lib/astarte_core/interface.ex
+  """
+  @spec interface() :: StreamData.t(%Interface{})
+  def interface() do
+    gen all(
+          required <- required_fields(),
+          optional <- optional_fields()
+        ) do
+      struct(Interface, Map.merge(required, optional))
+    end
+  end
+
+  defp id() do
+    repeatedly(&UUID.generate/0)
```

Comment:
we usually use binary uuids internally

```suggestion
    repeatedly(&UUID.bingenerate/0)
```

Reply:
The ID is an implementation detail: let's put it here because it's part of `%Astarte.Core.Interface{}`, but it should be removed from `Astarte.Core` sooner or later (probably later) as it's not part of the Interface logical entity